pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go

```diff
@@ -196,14 +197,14 @@ func NewAuthCreds(refHostname string) (AuthCreds, error) {
 			// When refHostname is "docker.io":
 			// - credFuncExpectedHostname: "registry-1.docker.io"
 			// - credFuncArg:              "registry-1.docker.io"
-			// - authConfigHostname:       "index.docker.io"
+			// - authConfigHostname:       "https://index.docker.io/v1/" (registry.IndexServer)
 			// - ac.ServerAddress:         "https://index.docker.io/v1/".
 			if !isAuthConfigEmpty(ac) {
 				if ac.ServerAddress == "" {
 					// This can happen with Amazon ECR: https://github.com/containerd/nerdctl/issues/733
 					logrus.Debugf("failed to get ac.ServerAddress for authConfigHostname=%q (refHostname=%q)",
 						authConfigHostname, refHostname)
-				} else {
+				} else if authConfigHostname != registry.IndexServer {
```

Comment:
What happens in the default case now?

Reply:
Nothing. The whole block is just an assertion anyways.

We could check that `ac.ServerAddress == registry.IndexServer` and return an error if it isn't. I guess I'm going to add it; it doesn't hurt.

Reply:
Done this now; makes the diff easier to understand too, so is a good thing.

Reply:
Sorry, I did one more change to move the check for `authConfigHostname == registry.IndexServer` all the way to the top to make sure it fails when `ac.ServerAddress == ""`, which should never happen for the IndexServer.